Sandbox/src/Game/Items/ItemRepository.cpp

```diff
@@ -0,0 +1,280 @@
+#include "Game/Items/ItemRepository.h"
+
+ItemRepository& ItemRepository::Instance()
+{
+	static ItemRepository instance;
+	return instance;
+}
+
+ItemRepository::ItemRepository()
+{
+	Init();
+}
+
+ItemRepository::~ItemRepository()
+{
+	for (auto item : mItemDefinitions)
+		delete item.second;
+
+	for (auto food : mFoodDefinitions)
+		delete food.second;
+
+	for (auto potion : mPotionDefinitions)
+		delete potion.second;
+
+	for (auto equipment : mEquipmentDefinitions)
+		delete equipment.second;
+
+	for (auto itemBonus : mItemBonusDefinitions)
+		delete itemBonus.second;
+
+	for (auto weapon : mWeaponDefinitions)
+		delete weapon.second;
+
+	for (auto ranged : mRangedDefinitions)
+		delete ranged.second;
+}
+
+void ItemRepository::Init()
+{
+	LoadItemDefinitions();
+	LoadFoodDefinitions();
+	LoadPotionDefinitions();
+	LoadEquipmentDefinitions();
+	LoadItemBonusDefinitions();
+	LoadWeaponDefinitions();
+	LoadRangedDefinitions();
+}
+
+void ItemRepository::LoadItemDefinitions()
+{
+	std::string fullPath = Square::AssetManager::Instance().GetBasePath();
+	fullPath.append("Assets/Definitions/");
+
+	std::ifstream infile(fullPath + "ItemDefinition.dat");
+	if (!infile) std::cerr << "ItemDefinition.dat could not be opened!" << std::endl;
+	else std::cout << "Loading item definitions..." << std::endl;
+
+	std::string name;
+	int index;
+	bool stackable;
+	int alchemyPrice;
+	int generalPrice;
+	std::string spritePath;
+
+	while (infile >> std::quoted(name) >> index >> stackable >> alchemyPrice >> generalPrice >> spritePath)
```

Comment:
what are these conditionals of `inflie >> std::quoted(name) >> etc.` doing?

Reply:
It's reading the file line by line. The while loop condition is if it can find those values in the line. And the ```>>``` stores the value from the text file into the variable.